fluss-dist/src/main/assemblies/assembly.xml

```diff
@@ -0,0 +1,157 @@
+<!--
+ Copyright (c) 2024 Alibaba Group Holding Ltd.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+<assembly>
+    <id>dist</id>
+    <formats>
+        <format>dir</format>
+        <format>tar.gz</format>
+    </formats>
+
+    <includeBaseDirectory>true</includeBaseDirectory>
+    <baseDirectory>fluss-${project.version}-bin</baseDirectory>
+
+    <dependencySets>
+        <dependencySet>
+            <outputDirectory>lib</outputDirectory>
```

Comment:
What's the different between this `assembly.xml` and the existing `bin.xml`, `opt.xml`, `plugins.xml`? We have build `fluss-xxx-bin` directory. Can we just add a phase to package the directory into a `.tar.gz` file?

Reply:
Hi,file `assembly.xml` is merged from `bin.xml`, `opt.xml` and `plugins.xml` with phase `tar.gz`, since adding the `tar.gz` phase to each file doesn't work, I add a new file and keep the original three files. If these three files are no longer required, these files will be deleted and keep `assembly.xml` instead,is it OK?

Reply:
We can't remove `bin.xml`, `opt.xml` and `plugins.xml`, because they are used to build `fluss-${project.version}-bin` directory, and the final release distribution tgz is built from this directory with additional licenses and notice. Currently, this works with the `create_binary_release.sh` script. 

To improve maintainability, we should eliminate the redundant assembly XML definitions, which complicate maintenance. Instead, we should focus on reusing the `fluss-dist/target/fluss-${project.version}-bin` directory for building the local tgz.


Here is how it works in my local, you can take this commit:
https://github.com/wuchong/fluss/commit/1cc0143a41755510f4557207c2703f70c060fe7b